src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1854,16 +1854,16 @@ MavlinkReceiver::handle_message_heartbeat(mavlink_message_t *msg)
 		mavlink_heartbeat_t hb;
 		mavlink_msg_heartbeat_decode(msg, &hb);
 
-		/* ignore own heartbeats, accept only heartbeats from GCS */
-		if (msg->sysid != mavlink_system.sysid && hb.type == MAV_TYPE_GCS) {
-
+		/* accept only heartbeats from GCS or Onboard computers */
+		if (hb.type == MAV_TYPE_GCS || hb.type == MAV_TYPE_ONBOARD_CONTROLLER) {
```

Comment:
Is there any reason we can't drop this check entirely now?

Reply:
What if there is a heartbeat from another drone (in a swarm network), charging station, or any other device? Should it be considered as a telemetry link?
I don't think so.